pgjdbc/src/test/java/org/postgresql/test/jdbc2/BatchExecuteTest.java

```diff
@@ -214,25 +214,24 @@ public void testClearPreparedEmptyBatch() throws Exception {
   }
 
   @Test
-  public void testSelectInBatch() throws Exception {
+  public void testSelectThrowsException() throws Exception {
     Statement stmt = con.createStatement();
 
     stmt.addBatch("UPDATE testbatch SET col1 = col1 + 1 WHERE pk = 1");
     stmt.addBatch("SELECT col1 FROM testbatch WHERE pk = 1");
     stmt.addBatch("UPDATE testbatch SET col1 = col1 + 2 WHERE pk = 1");
 
-    // There's no reason to Assert.fail
-    int[] updateCounts = stmt.executeBatch();
-
-    Assert.assertTrue("First update should succeed, thus updateCount should be 1 or SUCCESS_NO_INFO"
-            + ", actual value: " + updateCounts[0],
-        updateCounts[0] == 1 || updateCounts[0] == Statement.SUCCESS_NO_INFO);
-    Assert.assertTrue("For SELECT, number of modified rows should be either 0 or SUCCESS_NO_INFO"
-            + ", actual value: " + updateCounts[1],
-        updateCounts[1] == 0 || updateCounts[1] == Statement.SUCCESS_NO_INFO);
-    Assert.assertTrue("Second update should succeed, thus updateCount should be 1 or SUCCESS_NO_INFO"
-            + ", actual value: " + updateCounts[2],
-        updateCounts[2] == 1 || updateCounts[2] == Statement.SUCCESS_NO_INFO);
+    try {
+      stmt.executeBatch();
+      Assert.fail("Should raise a BatchUpdateException because of the SELECT");
```

Comment:
Please add explicit reference that clarifies why "select should raise BatchUpdateException"